Compiling EDM4hep against custom podio in LCG fails to honor `PODIO` environment variable

- Goal: I'm attempting to compile EDM4hep in LCG with a more recent version of podio than is present in LCG_101
 - OS version: LCG_101/x86_64-ubuntu2004-gcc9-opt
 - Compiler version: gcc9
 - Package version: ffcdb0b34766e444d145367966037852f7692431
 - Reproduced by (originally inside a GitHub action with [run-lcg-views](https://github.com/AIDAsoft/run-lcg-views)):
```console
#    - name: Build in LCG view
#      uses: aidasoft/run-lcg-view@v1
#      with:
#        release-platform: ${{ matrix.LCG }}
#        run: |

docker run --rm -it -v /cvmfs:/cvmfs:shared ghcr.io/aidasoft/ubuntu2004:latest /bin/bash
source /cvmfs/sft.cern.ch/lcg/views/LCG_101/x86_64-ubuntu2004-gcc9-opt/setup.sh
git clone https://github.com/AIDAsoft/podio
git clone https://github.com/key4hep/edm4hep

          # build podio
          cmake -B podio-build -S podio \
            -DCMAKE_INSTALL_PREFIX=podio-install \
            -DUSE_EXTERNAL_CATCH2=OFF
          make -C podio-build -j6 install
          
          # build EDM4HEP
          export PODIO=$PWD/podio-install
          cmake -B edm4hep-build -S edm4hep \
            -DCMAKE_INSTALL_PREFIX=edm4hep-install \
            -DUSE_EXTERNAL_CATCH2=OFF
          make -C edm4hep-build -j6 install
```
This produces the following output (with the cmake error explained by podio 0.13.0 in `/cvmfs/sft.cern.ch/lcg/views/LCG_101/x86_64-ubuntu2004-gcc9-opt/include/podio/podioVersion.h`; that's the reason for the custom podio install):
```console
-- The CXX compiler identification is GNU 9.4.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/local/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found nlohmann_json: /cvmfs/sft.cern.ch/lcg/views/LCG_101/x86_64-ubuntu2004-gcc9-opt/lib/cmake/nlohmann_json/nlohmann_jsonConfig.cmake (found suitable version "3.9.1", minimum required is "3.9.1") 
-- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.so (found version "1.2.11") 
-- Found SIO: /cvmfs/sft.cern.ch/lcg/views/LCG_101/x86_64-ubuntu2004-gcc9-opt/include  
-- Found podio: /cvmfs/sft.cern.ch/lcg/views/LCG_101/x86_64-ubuntu2004-gcc9-opt/lib/cmake/podio/podioConfig.cmake  
-- C++ standard: 17
-- Performing Test CXX_FLAG_WORKS__fPIC
-- Performing Test CXX_FLAG_WORKS__fPIC - Success
-- Adding -fPIC to CXX_FLAGS
-- Performing Test CXX_FLAG_WORKS__Wall
-- Performing Test CXX_FLAG_WORKS__Wall - Success
-- Adding -Wall to CXX_FLAGS
-- Performing Test CXX_FLAG_WORKS__Wextra
-- Performing Test CXX_FLAG_WORKS__Wextra - Success
-- Adding -Wextra to CXX_FLAGS
-- Performing Test CXX_FLAG_WORKS__Wpedantic
-- Performing Test CXX_FLAG_WORKS__Wpedantic - Success
-- Adding -Wpedantic to CXX_FLAGS
-- Performing Test CXX_FLAG_WORKS__Wno_unused_variable
-- Performing Test CXX_FLAG_WORKS__Wno_unused_variable - Success
-- Adding -Wno-unused-variable to CXX_FLAGS
-- Performing Test CXX_FLAG_WORKS__Wno_unused_parameter
-- Performing Test CXX_FLAG_WORKS__Wno_unused_parameter - Success
-- Adding -Wno-unused-parameter to CXX_FLAGS
-- Detected GNU compatible linker
Traceback (most recent call last):
  File "/cvmfs/sft.cern.ch/lcg/views/LCG_101/x86_64-ubuntu2004-gcc9-opt/python/podio_class_generator.py", line 444, in <module>
    gen = ClassGenerator(args.description, args.targetdir, args.packagename, args.iohandlers,
  File "/cvmfs/sft.cern.ch/lcg/views/LCG_101/x86_64-ubuntu2004-gcc9-opt/python/podio_class_generator.py", line 68, in __init__
    self.reader.read()
  File "/cvmfs/sft.cern.ch/lcg/releases/podio/00.13-02c84/x86_64-ubuntu2004-gcc9-opt/python/podio_config_reader.py", line 438, in read
    validator.validate(self.components, self.datatypes, self.options.get("exposePODMembers", False))
  File "/cvmfs/sft.cern.ch/lcg/releases/podio/00.13-02c84/x86_64-ubuntu2004-gcc9-opt/python/podio_config_reader.py", line 154, in validate
    self._check_components()
  File "/cvmfs/sft.cern.ch/lcg/releases/podio/00.13-02c84/x86_64-ubuntu2004-gcc9-opt/python/podio_config_reader.py", line 177, in _check_components
    raise DefinitionError("{} defines a member of type '{}' which is not allowed in components"
generator_utils.DefinitionError: a defines a member of type 'int32_t' which is not allowed in components
-- Not adding the ROOT dictionary to the targets because the selection.xml has not been generated
CMake Error at edm4hep/CMakeLists.txt:9 (add_library):
  add_library cannot create ALIAS target "edm4hep::edm4hepDict" because
  target "edm4hepDict" does not already exist.

-- Not adding the SIO Blocks library to the targets because the corresponding c++ sources have not been generated
CMake Error at edm4hep/CMakeLists.txt:19 (install):
  install TARGETS given target "edm4hepDict" which does not exist.

-- Found HepPDT: /cvmfs/sft.cern.ch/lcg/views/LCG_101/x86_64-ubuntu2004-gcc9-opt/include  
-- Fetching local copy of Catch2 library for unit-tests...
-- Configuring incomplete, errors occurred!
See also "/edm4hep-build/CMakeFiles/CMakeOutput.log".
```

Hinting at the location of podio with the environment variable `PODIO` does not do the job, despite the `find_package(podio REQUIRED HINTS $ENV{PODIO})` at https://github.com/key4hep/EDM4hep/blob/master/CMakeLists.txt#L15. Hinting with `-Dpodio_ROOT=podio-install` fails as well, as does `-Dpodio_DIR=podio-install`.

It appears that in all cases the environment variable `CMAKE_PREFIX_PATH` takes precedence, and only after `unset CMAKE_PREFIX_PATH` is the `PODIO` environment variable honored. However, that causes other packages not to be found (e.g. HepPDT), which are found above.

It would seem that package-specific environment variables should take precedence over general cmake prefix environment variables, and that specific command line definitions should take precedence over all environment variables.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.3)
+cmake_minimum_required(VERSION 3.16)
 
 project(EDM4HEP LANGUAGES CXX)
 

--- a/README.md
+++ b/README.md
@@ -30,8 +30,8 @@ This project is in a beta stage -- feedback and use of it in production is encou
 
 Required:
 
-* [PODIO](https://github.com/AIDASoft/podio) >= v00-09-02
-* [ROOT](https://github.com/root-project/root) >= v6.18/04
+* [PODIO](https://github.com/AIDASoft/podio) >= v00-15
+* [ROOT](https://github.com/root-project/root) >= v6.22/06
 
 Optional:
 

